src/librustdoc/json/conversions.rs

```diff
@@ -507,14 +507,19 @@ impl FromWithTcx<clean::Impl> for Impl {
             blanket_impl,
             span: _span,
         } = impl_;
+        // FIXME: should `trait_` be a Path in JSON?
+        let trait_ = trait_.map(|path| {
+            let did = path.res.def_id();
+            clean::ResolvedPath { path, did }.into_tcx(tcx)
+        });
```

Comment:
I did this to avoid bumping the format-version, but it would be better to cleanup the JSON type too. Should I do that and bump the format-version?